privacy-contracts/src/main/solidity/DefaultOnChainPrivacyGroupManagementContract.sol

```diff
@@ -12,54 +13,58 @@ contract DefaultOnChainPrivacyGroupManagementContract is OnChainPrivacyGroupMana
         return _version;
     }
 
-    // overrides
     function canExecute() external view returns (bool) {
         return _canExecute;
     }
 
     function lock() public {
         require(_canExecute);
+        require(tx.origin == _owner, "Origin not the owner.");
         _canExecute = false;
     }
 
     function unlock() public {
         require(!_canExecute);
+        require(tx.origin == _owner, "Origin not the owner.");
         _canExecute = true;
     }
 
-    function addParticipants(bytes32 _enclaveKey, bytes32[] memory _accounts) public returns (bool) {
+    function addParticipants(bytes32[] memory _accounts) public returns (bool) {
         require(!_canExecute);
-        if(distributionList.length == 0) {
-            addParticipant(_enclaveKey);
+        if (_owner == address(0x0)) {
+            _owner = tx.origin;
         }
-        require(isMember(_enclaveKey));
-        bool result = addAll(_enclaveKey, _accounts);
+        require(tx.origin == _owner, "Origin not the owner.");
+        bool result = addAll(_accounts);
         _canExecute = true;
         updateVersion();
         return result;
     }
 
-    function removeParticipant(bytes32 _enclaveKey, bytes32 _account) public returns (bool) {
-        require(isMember(_enclaveKey));
+    function removeParticipant(bytes32 _account) public returns (bool) {
+        require(_canExecute);
+        require(tx.origin == _owner, "Origin not the owner.");
         bool result = removeInternal(_account);
         updateVersion();
+        emit ParticipantRemoved(result, _account);
         return result;
     }
 
-    function getParticipants(bytes32 _enclaveKey) public view returns (bytes32[] memory) {
-        require(isMember(_enclaveKey));
+    function getParticipants() public view returns (bytes32[] memory) {
         return distributionList;
     }
 
+    function canUpgrade() external returns (bool) {
+        require(tx.origin == _owner, "Origin not the owner.");
```

Comment:
This seems a bit weird. Aren't we supposed to return true or false?

I feel like we could have: `return tx.origin == _owner`.

Am I missing something?

Reply:
is there an implicit return true in solidity?

Reply:
Good catch!